### PR TITLE
Fix for associating external target for protection policy

### PR DESCRIPTION
**Pull request by bijayprasadtiwari**
Added a new cmdlet Get-CohesityVault

### CODE REVIEW

File: src/Cohesity.Powershell/Scripts/Vault/Get-CohesityVault.ps1 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        $token = $session.Accesstoken.Accesstoken
+    }
+    Process { 
+        if ($VaultName) {
+            $url = $server + '/irisservices/api/v1/public/vaults?name=' + $VaultName
+        }
+        else {
+            $url = $server + '/irisservices/api/v1/public/vaults'
+        }
```
Comment (KavishreeShanmugam11) [inline src/Cohesity.Powershell/Scripts/Vault/Get-CohesityVault.ps1]: can we declare the url variable as common and append the name parameter if 'VaultName' exist?

File: src/Cohesity.Powershell/Scripts/ProtectionPolicy/New-CohesityProtectionPolicy.ps1 [hunk 2]
```diff
@@ -45,6 +50,28 @@ function New-CohesityProtectionPolicy {
...
+            $resp = Get-CohesityVault -VaultName $VaultName
+            if ($null -eq $resp) {
+                Write-Host "Vault (external target) name '$VaultName' does not exists"
+                return
+            }
+            # Except the vault info all the default values referenced from cohesity UI
+            $snapshotArchivalCopyPolicies = @(
+                @{
+                    copyPartial = $true
+                    daysToKeep  = 90
```
Comment (KavishreeShanmugam11) [inline src/Cohesity.Powershell/Scripts/ProtectionPolicy/New-CohesityProtectionPolicy.ps1]: Wont 'daysToKeep' value change based on 'RetainInDays' parameter value?

Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/ProtectionPolicy/New-CohesityProtectionPolicy.ps1]: For archival (on the vault) the retain value could be different from the snapshot protection in the cluster.

File: src/Cohesity.Powershell/Scripts/Vault/Get-CohesityVault.ps1 (hunk 1 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/Vault/Get-CohesityVault.ps1]: Fixed

**Pull request merged**